lib/fastly_nsq/rake_task.rb

```diff
@@ -34,19 +34,21 @@ def add_rake_task_description_if_one_needed
     end
 
     def run_tasks
+      listener_threads = []
+
       topics.each do |topic|
-        Thread.new do
+        listener_threads << Thread.new do
           wrap_helpful_output(topic) do
             MessageQueue::Listener.new(topic: topic, channel: channel).go
           end
         end
       end
 
-      non_main_threads.map(&:join)
-    end
-
-    def non_main_threads
-      (Thread.list - [Thread.main])
+      while listener_threads.any?(&:status)
+        listener_threads.each do |thread|
+          thread.join(1)
```

Comment:
Changed it to use `Thread.abort_on_exception=` as that seems less likely to confuse.